app/common/state/aboutPreferencesState.js

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this file,
+* You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const assert = require('assert')
+
+// utils
+const {makeImmutable, isMap} = require('../../common/state/immutableUtil')
+
+const validateState = (state) => {
+  state = makeImmutable(state)
+  assert.ok(isMap(state), 'state must be an Immutable.Map')
+  // assert.ok(isMap(state.getIn(['about', 'preferences'])), 'state must contain an Immutable.Map of \'about\' \'preferences\'')
+  return state
+}
+const aboutPreferencesState = {
+  setBackupStatus: (state, status) => {
+    state = validateState(state)
+    if (status == null) {
+      return state
+    }
+    const date = new Date().getTime()
+    state = state.setIn(['about', 'preferences', 'backupSucceeded'], status)
+    return state.setIn(['about', 'preferences', 'updatedStamp'], date)
+  },
+
+  hasBeenBackedUp: (state) => {
+    state = validateState(state)
+    return (state.getIn(['about', 'preferences', 'backupSucceeded']) || state.getIn(['about', 'preferences', 'recoverySucceeded'])) || false
+  },
+
+  getPreferencesProp: (state, key) => {
+    state = validateState(state)
+    return state.getIn(['about', 'preferences', key]) || 0
+  },
+
+  setPreferencesProp: (state, key, value) => {
+    state = validateState(state)
+    if (key == null || value == null) {
```

Comment:
I would here check only for key, because sometimes you would like to assign null to some prop in the state